LN-Zap/zap-desktop
# Extract reusable components into storybook

## Description:

Create a storybook in order to aid component development by providing an easy way to preview components in isolation. Build out library of reusable components.

**Key changes include:**

 - Setup [storybook framework](https://github.com/storybooks/storybook).
 - Build out modules to encapsulate our theme settings (dark/light)
 - Build out set of initial base components (icons, buttons, messages etc).

**Notable items:**

- I have used [svgr](https://github.com/smooth-code/svgr) to optimise and convert off of our custom svg icons into components that can be more easily used in a consistent way.
- `yarn generate-icons` will use svgr to automatically generate react components for each svg in the icons directory and place the resulting component into `app/components/Icon/`.
- `yarn storybook` with start up the storybook isolated development environment.
- `yarn storybook:deploy` will build the storybook and deploy to github pages.

## Motivation and Context:

- DX for working on the app could be improved by making it easy to work on components outside of the context of the electron app, which is slow to start up and hard to navigate to all places without having a wallet in a particular state.
- Make code more DRY and reusable.
- Fix design inconsistencies.

## How Has This Been Tested?

1. Start up the storybook `yarn storybook`.
2. Edit components and see them update in realtime in the storybook.
2. Deploy the storybook to git-pages with `yarn storybook:deploy`.
3. View the deployed storybook at https://ln-zap.github.io/zap-desktop/?no-cache=1.

## Screenshots (if appropriate):

<img width="1437" alt="screenshot 2018-10-04 21 18 30" src="https://user-images.githubusercontent.com/200251/46497510-13672880-c81b-11e8-85d2-084b3547d7f3.png">

## Types of changes:

New Feature

## Checklist:
